Adjust AssetGroupSighting detect API code

karenc opened the pull request:
- Create oauth user for acm before first request

  Add code to create oauth user for acm to use before first request.  ACM
  needs a oauth client id and secret in order to get the images from
  houston.  This adds a static client id and secret which is shared by
  houston and acm and it is created before the first request so ACM will
  be able to authenticate.

- Adjust AssetGroupSighting data sent to wbia detect API

  - Use `urllib.parse.urljoin` to create urls
  - Prepend `houston+` to urls that require oauth2 authentication
  - Change `image_uuid_list` to a list of asset urls
  - Move `callback_url` and `callback_detailed` from inside `input` to
    outside
  - Save asset ids with the job info in AssetGroupSighting

- Adjust code for AssetGroupSighting detection callback handler

  - Remove the extra level of API status code and message that's returned
    from normal sage APIs.  This works differently because it's a
    callback.
  - Disregard the image uuids returned from sage because those are
    different from the houston ones, the asset ids are stored with the job
    info
  - Update the tests with real results from sage

karenc commented:
@mmulich Could you look at the "Create oauth user for acm..." commit and see if that's an acceptable way to share oauth secrets?

karenc commented:
@hwindsor I changed the data we're sending to the detection API (and made the changes to wbia in https://github.com/WildMeOrg/wildbook-ia/pull/196).  Does it look ok?

hwindsor commented on .env:
```diff
@@ -53,6 +53,11 @@ EDM_AUTHENTICATIONS_PASSWORD__DEFAULT=4dm1n
+# OAuth client id and secret
+OAUTH_USER_EMAIL=oauth-user@wildme.org
+HOUSTON_CLIENT_ID=5ecba12d-d102-498e-805f-5531b448a67e
+HOUSTON_CLIENT_SECRET=6wql6OZ5TyRRDFFfC0SPhzFCyBXVQHVWOxgLBge24crbh4vuTQhOlTvedZ3Nay21
```
If this is checked into git, and visible on gitlab, surely it's not secret?

karenc replied:
That's the names given to the strings for creating oauth clients -- client id and secret.

Anyway, this file is for development, we also put in things like `GITLAB_ADMIN_PASSWORD` and database passwords.  They should all be changed to something else for deployment.

hwindsor commented:
> @hwindsor I changed the data we're sending to the detection API (and made the changes to wbia in [WildMeOrg/wildbook-ia#196](https://github.com/WildMeOrg/wildbook-ia/pull/196)). Does it look ok?

The changes look fine, but please update the section in the IA Pipeline doc that gives an example of the message.
https://docs.google.com/document/d/11TMq1qzaQxva97M3XYwaEYYUawJ5VsrRXnJvTQR_nG0/edit?pli=1#heading=h.eakedinkslxg

karenc commented:
> > @hwindsor I changed the data we're sending to the detection API (and made the changes to wbia in [WildMeOrg/wildbook-ia#196](https://github.com/WildMeOrg/wildbook-ia/pull/196)). Does it look ok?
> 
> The changes look fine, but please update the section in the IA Pipeline doc that gives an example of the message.
> https://docs.google.com/document/d/11TMq1qzaQxva97M3XYwaEYYUawJ5VsrRXnJvTQR_nG0/edit?pli=1#heading=h.eakedinkslxg

Updated

karenc commented:
/rebase

colinwkingen commented:
/rebase

hwindsor reviewed (approved):
LGTM,
